src/core/algorithms/ind/ind.h

```diff
@@ -13,10 +14,14 @@ class IND {
 private:
     std::shared_ptr<ColumnCombination> lhs_;
     std::shared_ptr<ColumnCombination> rhs_;
+    std::shared_ptr<ColumnNamesData> column_names_data_;
```

Comment:
Let's reuse existing class `RelationalSchema`, that describes a single relation. And we will simply use a vector of these schemas:
```suggestion
    std::shared_ptr<std::vector<RelationalSchema>> schemas_;
```
Initialization of schema looks like:
```cpp
    schema = std::make_shared<RelationalSchema>(input_table->GetRelationName());
    for (size_t col_id = 0; col_id < input_table->GetNumberOfColumns(); ++col_id) {
        schema->AppendColumn(input_table->GetColumnName(col_id));
    }
```